svd-rs/src/dimelement.rs

```diff
@@ -170,7 +170,11 @@ impl DimElement {
     pub fn indexes_as_range(&self) -> Option<RangeInclusive<u32>> {
         let mut integers = Vec::with_capacity(self.dim as usize);
         for idx in self.indexes() {
-            integers.push(idx.parse::<u32>().ok()?);
+            let val = idx.parse::<u32>().ok()?;
+            if val.to_string() != idx {
+                return None;
+            }
```

Comment:
what exactly does this fix?

that `01` != `1`?

Reply:
to note, I think it's the wrong way to fix the issue but maybe it's not

Reply:
> what exactly does this fix?
> 
> that `01` != `1`?

Yes

Reply:
can we add a comment here to explain why we need the name to be an exact match with the integer representation

Reply:
could you add comment yourself, please? I have issues with internet, maybe several days